index.js

```diff
@@ -67,52 +86,62 @@ export default function (options) {
       cleanupOutputDirectory(options?.outputWhitelist ?? [])
       moveOutputToServerFolder()
 
-      let env = {
+      const nodeServer = createNodeServer(options?.healthcheckRoute ?? true)
+      writeFileToOutput(nodeServer, 'node-server.cjs')
+
+      let defaultEnv = {
         ADDRESS_HEADER: 'x-forwarded-for',
         XFF_DEPTH: '1',
       }
-
-      if (options?.envInWebconfig ?? true) {
-        const envPath = path.resolve(process.cwd(), '.env')
-        if (fs.existsSync(envPath)) {
-          Object.assign(
-            env,
-            parse(fs.readFileSync(envPath, { encoding: 'utf-8' }))
-          )
-        }
-        console.info(`Included .env variables in web.config`)
-      } else {
-        console.info(`Didn't include .env variables in web.config (disabled)`)
-      }
-      for (const key in env) {
-        // XML attributes cannot contain these characters, will result in IIS Error 500.19
-        env[key] = env[key]
-          .replaceAll('"', '&quot;')
-          .replaceAll("'", '&apos;')
-          .replaceAll('<', '&lt;')
-          .replaceAll('>', '&gt;')
-          .replaceAll('&', '&amp;')
-      }
-
       if (typeof options.origin !== 'string') {
         console.warn(
           `sveltekit-adapter-iis: unspecified option 'origin'!\nForm actions will likely return errror 403: Cross-site POST form submissions are forbidden`
         )
       } else {
-        env.ORIGIN = options.origin
+        defaultEnv.ORIGIN = options.origin
       }
 
-      const webConfig = createWebConfig({
-        env: env,
-        nodePath: options?.overrideNodeExePath,
-        externalRoutes: options?.externalRoutes,
-        externalRoutesIgnoreCase: options?.externalRoutesIgnoreCase,
-        redirectToHttps: options?.redirectToHttps
-      })
-      const nodeServer = createNodeServer(options?.healthcheckRoute ?? true)
-
-      writeFileToOutput(webConfig, 'web.config')
-      writeFileToOutput(nodeServer, 'node-server.cjs')
+      for (const [envFn, stage] of getEnvs()) {
+				const wcFilename = stage ? `web.${stage}.config` : 'web.config'
+        const env = {}
+				if (wcFilename === 'web.config') Object.assign(env, defaultEnv)
+				
+				if (options?.envInWebconfig ?? true) {
+					const envPath = path.resolve(process.cwd(), envFn)
+					if (fs.existsSync(envPath)) {
+						Object.assign(
+							env,
+							parse(fs.readFileSync(envPath, { encoding: 'utf-8' }))
+						)
+					} else {
+						console.warn(
+							`Didn't include ${envFn} variables in ${wcFilename} (${envPath} does not exist!)`
+						)
+					}
+					console.info(`Included ${envFn} variables in ${wcFilename}`)
+				} else {
+					console.info(
+						`Didn't include ${envFn} variables in ${wcFilename} (disabled)`
+					)
+				}
+				// XML attributes cannot contain these characters, will result in IIS Error 500.19
+				for (const key in env) {
+					env[key] = xmlEscape(env[key])
+				}
+
```

Comment:
It looks like this is getting formatted strangely on github.  It looks like it's using hard-tabs.  Running prettier on it should fix it.